### PR TITLE
Dbkvs Ranged Tombstones Impl

### PR DESCRIPTION
**Pull request by gmaretic**
**Goals (and why)**:
Implement efficient ranged tombstones for Dbkvs, for targeted sweep.

**Implementation Description (bullets)**:
Pretty straightforward.

**Testing (What was existing testing like?  What have you done to improve it?)**:
There was decent test coverage, but added tests for edge cases.

**Concerns (what feedback would you like?)**:
Running benchmarks on postgres shows the expected huge perf improvement. Oracle does not seem to show improvement, so something may be wrong with the hinting in `OracleOverflowWriteTable`
 
**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:
Not super urgent, also want to get https://github.com/palantir/atlasdb/pull/3374 merged in before, so we can see the benchmarking delta

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3379)

### CODE REVIEW

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java [hunk 1]
```diff
@@ -1240,32 +1238,63 @@ public void deleteTimestampRangesIncludingSentinelsDeletesMultipleColumnsAcrossM
...
+        keyValueService.addGarbageCollectionSentinelValues(TEST_TABLE, ImmutableSet.of(TEST_CELL));
+        keyValueService.putWithTimestamps(TEST_TABLE, ImmutableMultimap.of(
+                TEST_CELL, Value.create(value0_t0, minTs),
+                TEST_CELL, Value.create(value0_t1, TEST_TIMESTAMP - 1),
+                TEST_CELL, Value.create(value0_t5, TEST_TIMESTAMP)));
+
+        keyValueService.deleteAllTimestamps(TEST_TABLE, ImmutableMap.of(TEST_CELL, TEST_TIMESTAMP), true);
+
+        assertThat(getAllTimestampsForTestCell(), contains(TEST_TIMESTAMP));
+    }
```
Comment (jeremyk-91) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: OK - tests make sense.

**Comment by tboam** [comment]
Do we understand why the results are bi-modal?  Is the DB caching the queries?  Is something in Java getting compiled and so runs faster?

**Comment by jeremyk-91** [comment]
@tboam I don't think we understand it yet. I have some bandwidth to run a few experiments today. 

**Caching**: We can validate this - if there is caching and we run the benchmark for longer, we shoulud expect to see the average time and the proportion of queries that are in the 'slow' group to decrease.

**JIT compilation**: Possible, though as far as I know the time taken in Atlas code is very small (it builds the SQL query and sends it over - we don't actually receive much other than the number of deleted elements (`9999`). I've also run the raw query a couple times in SQLPlus and have seen both fast and slow queries (e.g. <20ms to >2.7s).

**Comment by gmaretic** [comment]
@berler do you know why we would be seeing this behaviour on Oracle? Or do you have a suggestion on how to do things differently?

**Comment by gmaretic** [comment]
Should we merge this in as-is and reevaluate if needed later?
